### DISCUSSION TITLE
[PSD3] Exercise 5.33

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#132] [Esercizi (A-L)]
**Exercise 5.33** Express the following base 10 numbers in IEEE 754 single precision floating-point format. Express your answer in hexadecimal.

(a) $−13.5625$
(b) $42.3125$
(c) $−17.15625$

- _**N.B.:** I numeri da convertire sono gli stessi dell'Esercizio 5.29 (#130 ) e quindi anche dell'Esercizio 5.31 (#131 ), ma questa volta sono da esprimere in formato floating-point IEEE 754 single precision._

**Comment by Elia-Belli** [comment]
Convertiamo il numero da Decimale a Binario (come in fixed-point ma senza il bit del segno), riscriviamo in "notazione scientifica", convertiamo l'esponente (biased exponent), convertiamo in IEEE754. (indico con ... la mantissa se le cifre seguenti sono tutti 0)

(a) $-13.5625 = -1.1011001*2^3$ , exp = $127+3 = 130_{10}$ $\rightarrow 1\ 10000010\ 1011001... = C1590000_{16}$

(b) $42.3125 = 1.010100101*2^5$ , exp = $127+5 = 132_{10}$ $\rightarrow 0\ 10000100\ 010100101... = 42294000_{16}$

(c) $-17.15625 = -1.000100101*2^4$ , exp = $127+4 = 131_{10}$ $\rightarrow 1\ 10000011\ 000100101... = C1894000_{16} $

**Comment by FeddyLix17** [comment]
(a) a me viene 1 10000010 1011001000.. e quindi C159000 in base 16 
(b) giusto
(c) giusto

**Comment by Elia-Belli** [comment]
@FedVlogger17 Corretto grazie
